pkg/services/ngalert/state/historian/loki.go

```diff
@@ -263,7 +264,8 @@ func statesToStream(rule history_model.RuleMeta, states []state.StateTransition,
 			Condition:      rule.Condition,
 			DashboardUID:   rule.DashboardUID,
 			PanelID:        rule.PanelID,
-			InstanceLabels: removePrivateLabels(state.Labels),
+			Fingerprint:    labelFingerprint(sanitizedLabels),
```

Comment:
state contains cacheID, why don't you use it?

Reply:
alternative, state has a method GetAlertInstanceKey that contains a hash of the labels. The only difference is that it include private labels. Why do you remove private labels?

Reply:
See comments above, I think it's a much more portable solution to match with Prometheus's fingerprint calculation.

State history has always removed private labels for a very long time, even back when it wrote only annotations. The fingerprint should definitely be based off the labels that actually get written - otherwise, the result is confusing and won't align with fingerprints the UI might be calculating.

Whether we strip private labels _at all_ is another issue, making this behavior change everywhere is something we can consider - but such a change should be made separately from this PR. In this scope of this PR, the fingerprint should follow the labels we actually store.

Reply:
Not sure I understand, private labels contain information such as alert rule UID, by removing this label and calculating a hash based on only public ones you may encounter a situation when there are two rules that do not have any labels and produce the exact same set of labels would end up with exact same fingerprints.